socks/socks4.py

```diff
@@ -86,13 +93,17 @@ def request(
         if user_id is None:
             user_id = self.user_id or b""
 
+        RequestCls: typing.Union[
+            typing.Type[SOCKS4Request], typing.Type[SOCKS4ARequest]
+        ] = SOCKS4Request
         address_type, encoded_addr = encode_address(addr)
         if address_type == AddressType.IPV6:
             raise SOCKSError("IPv6 addresses not supported by SOCKS4")
-        elif address_type == AddressType.DN and not self.allow_domain_names:
-            raise SOCKSError("Domain names only supported by SOCKS4A")
+        elif address_type == AddressType.DN:
```

Comment:
I think we still want a way to distinguish between allowing or disallowing DN. From a consumer's perspective (like HTTPX) we are basically going to send url.host directly in, we want the proxy library to know what is and isn't allowed

Reply:
Yep, makes sense, forgot about the serving side of things 👍